plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -0,0 +1,195 @@
+import io.trino.spi.Page;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.TimeZoneKey;
+import io.trino.spi.type.TypeManager;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.TypeSignature.mapType;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Comparator.comparingLong;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeHistoryTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final List<CommitInfoEntry> commitInfoEntries;
+
+    public DeltaLakeHistoryTable(SchemaTableName tableName, List<CommitInfoEntry> commitInfoEntries, TypeManager typeManager)
+    {
+        requireNonNull(typeManager, "typeManager is null");
+        this.commitInfoEntries = ImmutableList.copyOf(requireNonNull(commitInfoEntries, "commitInfoEntries is null")).stream()
+                .sorted(comparingLong(CommitInfoEntry::getVersion).reversed())
+                .collect(toImmutableList());
+
+        tableMetadata = new ConnectorTableMetadata(
+                requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("version", BIGINT))
+                        .add(new ColumnMetadata("timestamp", TIMESTAMP_TZ_MILLIS))
+                        .add(new ColumnMetadata("user_id", VARCHAR))
+                        .add(new ColumnMetadata("user_name", VARCHAR))
+                        .add(new ColumnMetadata("operation", VARCHAR))
+                        .add(new ColumnMetadata("operation_parameters", typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("job", RowType.rowType(
+                                RowType.field("job_id", VARCHAR),
+                                RowType.field("job_name", VARCHAR),
+                                RowType.field("run_id", VARCHAR),
+                                RowType.field("job_owner_id", VARCHAR),
+                                RowType.field("trigger_type", VARCHAR))))
+                        .add(new ColumnMetadata("notebook", RowType.rowType(
```

Comment:
do we really need all these fields? like notebook and notebook id?

Reply:
Probably not. Removing this field from the table.